qemu/tests/migration_multi_host_with_stress.py

```diff
@@ -0,0 +1,90 @@
+def run(test, params, env):
+    """
+    KVM multi-host migration test:
+
+    Migration execution progress is described in documentation
+    for migrate method in class MultihostMigration.
+    steps:
+        1) boot vm and login, load stress in guest
+        2) do migration
+        3) wait for a while and stop stress
+        4) login guest when migrate done
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    preprocess_env = params.get("preprocess_env", "yes") == "yes"
+    mig_protocol = params.get("mig_protocol", "tcp")
+    mig_type = utils_test.qemu.MultihostMigration
+    if mig_protocol == "fd":
+        mig_type = utils_test.qemu.MultihostMigrationFd
+    if mig_protocol == "exec":
+        mig_type = utils_test.qemu.MultihostMigrationExec
+    if "rdma" in mig_protocol:
+        mig_type = utils_test.qemu.MultihostMigrationRdma
+
+    class TestMultihostMigration(mig_type):
+
+        def __init__(self, test, params, env):
+            super(TestMultihostMigration, self).__init__(test, params, env,
+                                                         preprocess_env)
+            self.srchost = self.params.get("hosts")[0]
+            self.dsthost = self.params.get("hosts")[1]
+            self.is_src = params["hostid"] == self.srchost
+            self.vms = params["vms"].split()
+            self.login_timeout = int(params.get("login_timeout", 360))
+            self.need_cleanup = params.get("need_cleanup", "yes") == "yes"
+
+        def migration_scenario(self):
+            def clean_up(vm):
+                kill_bg_stress_cmd = params.get("kill_bg_stress_cmd",
+                                                "killall -9 stress")
+
+                logging.info("Kill the background test in the guest.")
+                session = vm.wait_for_login(timeout=self.login_timeout)
+                session.sendline(kill_bg_stress_cmd)
+                session.close()
+
+            @error.context_aware
+            def start_worker(mig_data):
+                logging.info("Try to login guest before migration test.")
+                wait_bg_timeout = int(params.get("wait_bg_timeout", 30))
+                vm = env.get_vm(params["main_vm"])
+                session = vm.wait_for_login(timeout=self.login_timeout)
+
+                error.context("Do stress test before migration.", logging.info)
+                bg_stress_test = params.get("bg_stress_test")
+
+                bg = utils.InterruptedThread(utils_test.run_virt_sub_test,
+                                             args=(test, params, env,),
+                                             kwargs={"sub_type": bg_stress_test})
+
+                bg.start()
+                if not bg.is_alive():
+                    raise error.TestFail("Failed to start guest test!")
+                # waiting background test start.
+                time.sleep(wait_bg_timeout)
```

Comment:
for autotest.stress, when bg.start(), then bg is alive, but stress process is not begin, bg.start() include scp code, make and exec stress, so need wait some seconds for stress process is running.